### PR TITLE
CliMenu version bump, and TextInputOption for Menu

### PR DESCRIPTION
**Pull request by ijpatricio**
Hey @nunomaduro !

So, basically this PR is two subjects:

#### 1. CliMenu version bump
- CliMenu version bump, that changed namespace [Check](https://github.com/nunomaduro/laravel-console-menu/compare/master...ijpatricio:ijpatricio-feat-input-prompt?expand=1#diff-d1b2b70dc450f42ab1aed7b47ed90cfeL17)

#### 2. TextInputOption for Menu
- Added public method
`addTextInputOption(string $label, string $promptText, string $placeholder = 'Enter text...'): Menu`

    Consequence of this new feature, that returns only the user written string, also created a new class `InputTextOption`, to represent it correctly.

    Of course, this and the existing `MenuOption.php` implement the added Interface `SelectedOption.php`, that only enforces to implement the method `getValue()` that returns mixed. (existing one was `string|int`)

#### Finally

- Updated Changelog, hope all ok

- Added section on readme, with picture, to document how TextInput may be used, and also demonstrate to use text values (existing one was integers). [Check](https://github.com/ijpatricio/laravel-console-menu/tree/ijpatricio-feat-input-prompt#setup-with-text-values-and-user-input)

### CODE REVIEW

File: src/SelectedOption.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * (c) Nuno Maduro <enunomaduro@gmail.com>
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ */
+
+namespace NunoMaduro\LaravelConsoleMenu;
+
+interface SelectedOption
```
Comment (nunomaduro) [inline src/SelectedOption.php]: Good idea! I do think tho that would be better rename this interface to `OptionContract`. We should also place this interface under the contracts folder.

File: src/MenuOption.php [hunk 2]
```diff
@@ -16,9 +16,9 @@
 use PhpSchool\CliMenu\MenuItem\SelectableItem;
 
 /**
- * This is an Laravel Console Menu Option implementation.
+ * This is a Laravel Console Menu Option implementation.
  */
-class MenuOption extends SelectableItem
+class MenuOption extends SelectableItem implements SelectedOption
```
Comment (nunomaduro) [inline src/MenuOption.php]: Since now we may have more than one option we should rename `MenuOption`  to `Option` only.

File: src/Menu.php [hunk 3]
```diff
@@ -84,6 +86,32 @@ public function addOptions(array $options): Menu
...
 
+    /**
+     * Adds a new text input option.
+     *
+     * @param string $label
+     * @param string $promptText
+     * @param string $placeholder
+     * @return Menu
+     */
+    public function addTextInputOption(string $label, string $promptText, string $placeholder = 'Enter text...'): Menu
```
Comment (nunomaduro) [inline src/Menu.php]: Do you agree that, in the end, this is a question right?

What do you think about `addQuestion` instead?

Shouldn't the placeholder be null by default?

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [2.0.0] - 2018-07-30
+### Changed
+- Using `"php-school/cli-menu": "^3.0"`
+- Update namespace on CliMenuBuilder
```
Comment (nunomaduro) [inline CHANGELOG.md]: This changed topic can be only: "Upgraded Cli Menu to v3.".

**Review by nunomaduro — CHANGES_REQUESTED** [review]
Thanks, @ijpatricio for your work. I have made some comments, tell me what do you think about them?

Have you time this week to update the PR? Otherwise, I can do it for you.

File: CHANGELOG.md (hunk 4 above)
Comment (ijpatricio) [reply CHANGELOG.md]: Done.

File: src/Menu.php (hunk 3 above)
Comment (ijpatricio) [reply src/Menu.php]: Agreed.
Null not possible (respecting interface), so I put empty string.

File: src/MenuOption.php (hunk 2 above)
Comment (ijpatricio) [reply src/MenuOption.php]: As spoken on slack, kept it simple, removing the interface.

File: src/SelectedOption.php (hunk 1 above)
Comment (ijpatricio) [reply src/SelectedOption.php]: As spoken on slack, kept it simple, removing the interface.

**Comment by ijpatricio** [comment]
Hey @nunomaduro , no problem, thank you for your guidance!

So, just commited after our comments, hope all good!! 👍

**Pull request merged**